From 3468c0c1c6f0cef895ccd4bc167ceb7490b4f1ee Mon Sep 17 00:00:00 2001
From: "Simental Magana, Marcos" <marcos.simental.magana@intel.com>
Date: Wed, 15 Jan 2020 09:36:21 -0600
Subject: [PATCH] generate otb files

new pango (1.44.x) switched to use harfbuzz (which does not
support bitmap fonts). This change generates converted bitmap
fonts to OpenType Bitmap (otb) which are supported.

The patch is stolen from Fedora[1] and applied in git format

[1]-https://src.fedoraproject.org/rpms/terminus-fonts/raw/3a28b472a0a6b0a98b3982e7f201b5df72b8c6c8/f/terminus-font-opentype-bitmap-via-fontforge.patch

Signed-off-by: Simental Magana, Marcos <marcos.simental.magana@intel.com>
---
 Makefile | 25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d3b1d12..437a252 100644
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+FONTFORGE = fontforge
+INSTALL = install
 INT = python3
 EXT = py
 BIN = ./bin
@@ -211,6 +213,27 @@ install-psf-ref: README
 uninstall-psf-ref:
 	rm -f $(DESTDIR)$(psfref)
 
+# OpenType Bitmap fonts (OTB)
+
+OTB = ter-u12b.otb ter-u12n.otb ter-u14b.otb ter-u14n.otb ter-u16b.otb ter-u16n.otb ter-u18b.otb ter-u18n.otb ter-u20b.otb ter-u20n.otb ter-u22b.otb ter-u22n.otb ter-u24b.otb ter-u24n.otb ter-u28b.otb ter-u28n.otb ter-u32b.otb ter-u32n.otb
+
+%.otb: %.bdf
+	$(FONTFORGE) -lang=ff -c 'Open("$<"); Generate("$@")'
+
+.PHONY: otb
+otb: $(OTB)
+
+otbdir = $(x11dir)
+
+.PHONY: install-otb
+install-otb: $(OTB)
+	$(INSTALL) -m 0755 -d "$(DESTDIR)$(otbdir)"
+	$(INSTALL) -m 0644 -t "$(DESTDIR)$(otbdir)" $(OTB)
+
+.PHONY: uninstall-otb
+uninstall-otb:
+	@set -x; if test -d "$(DESTDIR)$(otbdir)"; then cd "$(DESTDIR)$(otbdir)" && rm -f $(OTB); fi
+
 # X11 Window System
 
 X11_8859_1  = uni/x11gr.uni uni/ascii-h.uni uni/win-1252.uni
@@ -291,6 +314,6 @@ uninstall-pcf-8bit:
 # Cleanup
 
 clean:
-	rm -f $(PSF) $(PSF_VGAW) $(PCF) $(PCF_8BIT) $(FNT)
+	rm -f $(PSF) $(PSF_VGAW) $(PCF) $(PCF_8BIT) $(FNT) $(OTB)
 
 .PHONY: all install uninstall fontdir psf install-psf uninstall-psf psf-vgaw install-psf-vgaw uninstall-psf-vgaw install-psf-ref uninstall-psf-ref pcf install-pcf uninstall-pcf pcf-8bit install-pcf-8bit uninstall-pcf-8bit clean
-- 
2.24.1

